api/envoy/config/filter/http/gzip/v2/gzip.proto

```diff
@@ -55,19 +60,33 @@ message Gzip {
   // application/json, text/html, etc. When this field is not defined, compression will be applied
   // to the following mime-types: "application/javascript", "application/json",
   // "application/xhtml+xml", "image/svg+xml", "text/css", "text/html", "text/plain", "text/xml".
-  repeated string content_type = 6 [(validate.rules).repeated = {max_items: 50}];
+  // .. attention:
+  //
+  //    **This field is deprecated**. Set the `compressor` field instead.
+  repeated string content_type = 6 [deprecated = true];
 
   // If true, disables compression when the response contains an etag header. When it is false, the
   // filter will preserve weak etags and remove the ones that require strong validation.
-  bool disable_on_etag_header = 7;
+  // .. attention:
+  //
+  //    **This field is deprecated**. Set the `compressor` field instead.
+  bool disable_on_etag_header = 7 [deprecated = true];
 
   // If true, removes accept-encoding from the request headers before dispatching it to the upstream
   // so that responses do not get compressed before reaching the filter.
-  bool remove_accept_encoding_header = 8;
+  // .. attention:
+  //
+  //    **This field is deprecated**. Set the `compressor` field instead.
+  bool remove_accept_encoding_header = 8 [deprecated = true];
 
   // Value from 9 to 15 that represents the base two logarithmic of the compressor's window size.
   // Larger window results in better compression at the expense of memory usage. The default is 12
   // which will produce a 4096 bytes window. For more details about this parameter, please refer to
   // zlib manual > deflateInit2.
   google.protobuf.UInt32Value window_bits = 9 [(validate.rules).uint32 = {lte: 15 gte: 9}];
+
+  // Set of configuration parameters common for all compression filters. If this field is set then
+  // the fields `content_length`, `content_type`, `disable_on_etag_header` and
+  // `remove_accept_encoding_header` are ignored.
+  compressor.v2.Compressor compressor = 10;
```

Comment:
Should this be a common message in a neutral common location? As it is today, I think we are pointing the decompressor at the compressor, but long term the compressor configuration may have diverging fields as well.

Reply:
Probably I misunderstood you comment. Currently decompression is not supported in Envoy.

I experimented a bit with symmetric compression/decompression for gzip and brotli and ended up with two different generalized filters: [one for compression and another for decompression](https://github.com/envoyproxy/envoy/issues/4445#issuecomment-501634446). The experimental branch introduces a new message for [`Decompressor`](https://github.com/rojkov/envoy/blob/decompress/api/envoy/config/filter/http/decompressor/v2/decompressor.proto) and I don't expect it to have common fields with `Compressor`. E.g. the `content_length` field doesn't make sense for decompressors for sure as well as `remove_accept_encoding_header`.

Reply:
Ack, discard this thread then.